lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -1011,9 +1016,15 @@ ViewListDiffResult? diffViewList(List<int> active, List<int> next) {
         return null;
       }
     }
+
+    // Remove all ids from viewsToRemove that also exist in viewsToAdd.
```

Comment:
Not entirely sure how to trigger/test this case, but looking at how the code [accesses the `viewClipChain`](https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart#L507), this branch of the code seems also vulnerable to the same error :/